sbkeys/README.md

```diff
@@ -0,0 +1,91 @@
+
+To streamline the process of building Bottlerocket with Secure Boot support, a local profile will be generated automatically.
+This is done to minimize costs and to avoid requiring developers to set up infrastructure for key management ahead of time.
+However, because these local profiles offer direct access to private key materials, they are **strongly discouraged** for any kind of production use.
+
+Different profiles can be [specified at build time](#specify-a-profile-at-build-time) to use a custom set of keys.
+
+## Create a profile with local resources
+
+The `generate-local-sbkeys` tool can be used to create a local Secure Boot Keys profile.
+
+It uses `openssl` to generate private keys and certificate authorities (CAs), and `gpg` to create a GPG private key.
+It also uses [virt-fw-vars](https://pypi.org/project/virt-firmware/) to generate EFI variable data for the edk2 variable stores used by QEMU, and for registering Amazon EC2 AMIs.
+
+```shell
+ARCH="$(uname -m)"
+SDK_VERSION="v0.29.0"
+./generate-local-sbkeys \
+  --sdk-image "public.ecr.aws/bottlerocket/bottlerocket-sdk-${ARCH}:${SDK_VERSION}" \
+  --output-dir "${PWD}/my-local-profile"
+```
+
+## Create a profile with AWS-based resources
+
+The `generate-aws-sbkeys` tool can be used to create an AWS-based Secure Boot Keys profile.
+
+It uses the AWS CLI and [aws-kms-pkcs11](https://github.com/JackOfMostTrades/aws-kms-pkcs11) to obtain certificates, and the `virt-fw-vars` tool to generate EFI variable data.
+It creates an `aws-kms-pkcs11` configuration file for subsequent signing operations.
+
+The tool expects four [AWS Private CAs](https://docs.aws.amazon.com/privateca/latest/userguide/PcaWelcome.html) and three [AWS KMS asymmetric keys](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#asymmetric-keys-concept) to be available.
+Note that the cost of these resources, **especially the private CAs**, is nontrivial.
+
+Although it is possible to use the same private CA and the same KMS key for all roles, doing so would weaken the security of the implementation, and is **strongly discouraged**.
+
+```shell
+ARCH="$(uname -m)"
+SDK_VERSION="v0.29.0"
+
+# AWS Private CAs
+PK_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/11111111-1111-1111-1111-111111111111"
+KEK_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/22222222-2222-2222-2222-222222222222"
+DB_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/33333333-3333-3333-3333-333333333333"
+VENDOR_CA="arn:aws:acm-pca:us-west-2:999999999999:certificate-authority/44444444-4444-4444-4444-444444444444"
+
+# AWS KMS asymmetric keys
+SHIM_SIGN_KEY="arn:aws:kms:us-west-2:999999999999:key/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
+CODE_SIGN_KEY="arn:aws:kms:us-west-2:999999999999:key/bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb"
+CONFIG_SIGN_KEY="arn:aws:kms:us-west-2:999999999999:key/cccccccc-cccc-cccc-cccc-cccccccccccc"
+
+./generate-aws-sbkeys \
```

Comment:
> Should pubsys be able to do this instead of a script? If so, wouldn't we want it to be possible to set and forget this by pointing Infra.toml at the right resources?

Conceptually, this script is somewhat closer to what we were trying to do with `infrasys`, where we create assets that can later be referenced in `Infra.toml`.

But I didn't see a good reason to try to standardize an "aws" shape and a "local" shape in `Infra.toml`, since the two profile types are very different and require different handling in `rpm2img` for signing to actually work.

`rpm2img` eventually does channel them into a mostly-shared code path, which requires correct NSS and GPG configurations so that `pesign` and `gpg` can "just work" later on. Even there, different arguments are required for the actual CLI invocation when signing.

What would help is if there were some sort of virtual smartcard support for local keys generated by OpenSSL. Then we could use the PKCS#11 code path consistently, and `pubsys-setup` or some other helper could have "configure smartcard for signing" as its contract, and `rpm2img` could just rely on a correctly-configured smartcard interface with standard key and token names.

AFAICT we'd have to write our own PKCS#11 provider with a "local" backend for that to work - like `aws-kms-pkcs11` does for "aws" - which was more than I wanted to tackle here. With that provider in place, we'd need something like these scripts to generate the required files. The output would just look more like this:
```
❯ tree aws local
aws
├── config-sign.key
├── kms-sign.json
└── efi-certs.json
local
├── config-sign.key
├── local-sign.json
└── efi-certs.json
```

We could shift some of what the call to `virt-fw-vars` does into an implementation where `pubsys-setup` reads cert information from `efi-certs.json` and creating `efi-vars.json`.

 I'm happy to backlog an issue for it; I could see it being useful for OOTBs for the same reason that it's useful that `tuftool` can abstract a bit over "local keys" vs. "KMS keys".